src/cargopacket.h

```diff
@@ -44,10 +44,16 @@ struct CargoPacket : CargoPacketPool::PoolItem<&_cargopacket_pool> {
 
 	Money feeder_share{0}; ///< Value of feeder pickup to be paid for on delivery of cargo.
 
-	TileIndex source_xy{0}; ///< The origin of the cargo.
+	TileIndex source_xy{INVALID_TILE}; ///< The origin of the cargo.
+	TileIndexDiffC cargo_movement{0, 0}; ///< A vector (from 0,0) of how far the cargo got actually transported in a vehicle. Is a vector when in station. Is a temporary variable when in vehicle.
```

Comment:
`cargo_` prefix is redundant imo as it a member of CargoPacket. You don't have `cargo_source_id` or `cargo_count`, so why `cargo_movement`?

Reply:
In contrast to the 'source' prefixes. But I also didn't really like it, and I would like 'vector" in the name. 'traveled_vector' maybe.